multiversx/mx-chain-go
# Fix the `changeOwner` functionality in the system delegation contract

## Reasoning behind the pull request
- the owner field was not updated when the `changeOwner` operation was processed on a system delegation contract
  
## Proposed changes
- added a fix for the change owner functionality on the delegation system SC

## Testing procedure
- normal system test. Before the `MultiClaimOnDelegationEnableEpoch` (epoch 3) perform a changeOwner operation. Check the account of the system delegation SC (using the /address/ endpoint). The owner should be the original one. After activation, call the `synchronizeOwner` operation in a new transaction on that delegation contract with no extra parameters. Then check again the user account, the correct owner should be in-place. Re-do a changeOwner operation, the correct owner should be found when querying the /address/ endpoint.

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
